### ISSUE TITLE
Githook for auto format 

### ISSUE THREAD

**Issue by leobenkel** [leobenkel/safety_plugin#42]
auto format and other cool stuff before git push

**Comment by brbrown25** [comment]
The following are some things that pop into my head that might be interesting to follow up on.
https://github.com/randomcoder/sbt-git-hooks
https://github.com/sbt/sbt-git
https://blog.softwaremill.com/reusable-pre-commit-hooks-in-scala-projects-d8bb327047ee

I think it might be this https://github.com/streamich/git-cz but in our node projects we have something that basically inits the git hooks for you. so would be cool if there was an sbt task you could run like sbt init and it would generate the pre commit hooks. just some thoughts I had following your awesome talk!

**Comment by leobenkel** [comment]
yes ! Absolutely ! This is exactly what I had in mind, or even hooking the task to the `reload` step of SBT so it runs each time you open up the sbt terminal. 

thank you for all the links ! Ill take a look and give a go at it as soon as i have free time.

**Comment by brbrown25** [comment]
I also would be down to give this a look when I have some free time as well

**Comment by leobenkel** [comment]
that would be amazing ! 

The more contributors, different point of views and use cases we have, the better !
